hudi-spark/src/main/java/org/apache/hudi/keygen/KeyGenerator.java

```diff
@@ -41,7 +41,7 @@
   private static final String STRUCT_NAME = "hoodieRowTopLevelField";
   private static final String NAMESPACE = "hoodieRow";
 
-  protected transient TypedProperties config;
+  protected  TypedProperties config;
```

Comment:
This is intended to be transient so that we don't  serialize config in KeyGenerator s. Lets keep it that way. Since the issue is specifically with CustomKeyGenerator we should resolve there.

Reply:
Ok i understand what you think, let me try other solutions

Reply:
Can you let me know what you think? Can you describe it in detail @bhasudha

Reply:
I think so too. @liujinhui1994  can you try with Hudi 0.6.0 and see if that helps? We fixed some serialization issues there wrt KeyGenerators.

Reply:
> I think so too. @liujinhui1994 can you try with Hudi 0.6.0 and see if that helps? We fixed some serialization issues there wrt KeyGenerators.

I tried Both `TimestampBasedKeyGenerator` and `CustomKeyGenerator` with 0.6.0 (in my IDEA) `TimestampBasedKeyGenerator` is ok, while `CustomKeyGenerator` still has NPE. 
Remove `transient ` can resolve this problem.

`Caused by: java.lang.NullPointerException
	at org.apache.hudi.keygen.SimpleKeyGenerator.<init>(SimpleKeyGenerator.java:35)
	at org.apache.hudi.keygen.CustomKeyGenerator.getRecordKey(CustomKeyGenerator.java:128)
	at org.apache.hudi.keygen.BuiltinKeyGenerator.getKey(BuiltinKeyGenerator.java:75)
	at org.apache.hudi.HoodieSparkSqlWriter$$anonfun$1.apply(HoodieSparkSqlWriter.scala:143)
	at org.apache.hudi.HoodieSparkSqlWriter$$anonfun$1.apply(HoodieSparkSqlWriter.scala:139)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
`

Reply:
@bhasudha @vinothchandar what was the purpose behind keeping this config object here as transient? There is a workaround solution also which only affects CustomKeyGenerator class but removing transient keyword from here actually seems the easier solution. :)

Reply:
If we do not want to make config object as transient, we have 2 choices - 
1. introduce a new TypedProperties instance in CustomKeyGenerator
2. introduce a new TimestampBasedKeyGenerator instance in CustomKeyGenerator, which gets initialised only if needed. 

Please suggest.